pb/Makefile

```diff
@@ -0,0 +1,17 @@
+PB = $(wildcard *.proto)
+GO = $(PB:.proto=.pb.go)
+
+ifeq ($(OS),Windows_NT)
+  GOPATH_DELIMITER = ;
```

Comment:
Pedantic:
GOPATH_DELIMITER -> LIST_SEPARATOR or something like that.

Reply:
Does this actually fix the build?

Reply:
It should https://github.com/libp2p/go-libp2p-pubsub-router/pull/33#issuecomment-520097219
Using the `:` causes this error.
`C:\Users\Dominic Della Valle\Projects\Go/src:.: warning: directory does not exist.`
Caveat: I am specifically using the official protobuf release for this because of this issue
https://github.com/msys2/MSYS2-packages/issues/1715

Reply:
Ah, I must have misunderstood the whole issue. I thought protoc was already splitting on `:`. Given that, it looks like `protoc` _correctly_ uses the OS path separator, right? In that case we could even just pass the `--protoc_path` flag twice.

Reply:
>In that case we could even just pass the --protoc_path flag twice.

Yes. However there was this
https://github.com/libp2p/go-libp2p/issues/700

>Ah, I must have misunderstood the whole issue. 

The difference in `protoc` binaries is what caused a lot of this confusion on our end as well.
What it looks like is that if you build protoc (at least v3.7.1) in a `mingw` environment, it has different parsing behavior, which is probably an upstream (pre-processor) issue.

The mingw build uses `:` and therefor will never work with Windows drive-spec paths since that arg needs an absolute path.
However, a relative path will work (if you're on that volume)
`msys2-protoc.exe --proto_path="\Users\Dominic Della Valle\Projects\Go/src:." --gogofast_out=. message.proto` succeeds

While it doesn't in mainline with either separator.
```
mainline-protoc --proto_path="\Users\Dominic Della Valle\Projects\Go/src;." --gogofast_out=. message.proto
\Users\Dominic Della Valle\Projects\Go/src: warning: directory does not exist.

mainline-protoc --proto_path="\Users\Dominic Della Valle\Projects\Go/src:." --gogofast_out=. message.proto
\Users\Dominic Della Valle\Projects\Go/src:.: warning: directory does not exist.
message.proto: File does not reside within any path specified using --proto_path (or -I).  You must specify a --proto_path which encompasses this file.  Note that the proto_path must be an exact prefix of the .proto file names -- protoc is too dumb to figure out when two paths (e.g. absolute and relative) are equivalent (it's harder than you think).
```

😵

Reply:
> Yes. However there was this

Ah. I responded to that _assuming_ we'd need to make an upstream change. Otherwise, meh.

> While it doesn't in mainline with either separator.

Ok, so, the solution is: use a separator _and_ resolve the full path _and_ don't use mingw.

Reply:
That sounds right. 
In regards to using 2 parameters or using 1 with the separator. The only concern is someone combining them later without realizing it will break building on Windows. But that's being extra defensive so I'm fine dropping that.

>and don't use mingw.

I plan to report this upstream to protobuf if I don't hear anything on the MSYS issue by end of day Monday.
It's hard to know if this is expected behavior or not, as well as if it was resolved already in later versions.
So it will either require a version bump and rebuild for the MSYS2 repositories, a patch upstream, or an explicit "not a bug" from upstream.

Reply:
I believe this is an MSYS issue. Unless I'm mistaken, `protoc` is being built as a _Linux_ program so it expects _Linux_ paths.

I think the right solution there is to pass `/c/...` (or just run `cygpath -u $YOUR_PATH`).

Reply:
It looks like you're right.
`msys2-protoc.exe --proto_path="/c/Users\Dominic Della Valle\Projects\Go/src:." --gogofast_out=. message.proto` succeeds (even with the mixed slash mess)

I might have skipped over that when testing. I remember @aschmahmann trying paths that looked like cygwin style paths without success, but that may have been when using the Linux ELF inside of WSL.

Reply:
> GOPATH_DELIMITER -> LIST_SEPARATOR

Done 😄

Reply:
> I remember @aschmahmann trying paths that looked like cygwin style paths without success, but that may have been when using the Linux ELF inside of WSL.

WSL does linux paths, and needs `/mnt/c/Users...`. We did some experimentation with skipping the drive letter and using `/Users/...` and I think that only really worked with MSYS.